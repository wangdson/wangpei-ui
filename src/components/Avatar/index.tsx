import React, { useMemo } from 'react';
import { glow } from '../shared/animation';
import {Icon} from '../Icon';
import { color, typography } from '../shared/style';
import styled, { css } from 'styled-components';

export const AvatarSize = {
  tiny: 16,
  small: 20,
  medium: 28,
  large: 40,
}

export interface AvatarProps {
  /** 是否加载中 */
  isLoading?: boolean,
  /** 用户名 */
  username?: string,
  /** button大小 */
  size?: keyof typeof AvatarSize,
  /** 图片地址 */
  src?: null|string,
}

const Image = styled.div<AvatarProps>`
  background: ${props => (!props.isLoading? 'transparent': color.lightest)};
  border-radius: 50%;
  display: inline-block;
  vertical-align: top;
  overflow: hidden;
  text-transform: uppercase;
  height: ${props => AvatarSize[props.size as keyof typeof AvatarSize]}px;
  width: ${props => AvatarSize[props.size as keyof typeof AvatarSize]}px;
  line-height: ${props => AvatarSize[props.size as keyof typeof AvatarSize]}px;
  img {
    width: 100%;
    height: auto;
    display: block;
  }
  svg {
    position: relative;
    bottom: -2px;
    height: 100%;
    width: 100%;
    vertical-align: top;
  }
  path {
    fill: ${color.medium},
    animation: ${glow} 1.5s ease-in-out infinite;
  }
`;

const Initial = styled.div<AvatarProps>`
  background: ${color.darkest};
  color: ${color.lightest};
  text-align: center;
  font-size: ${typography.size.s2}px;
  line-height: ${AvatarSize.medium}px;
  ${props => props.size === 'tiny' && css`
    font-size: ${parseFloat(typography.size.s1) - 2}px;
    line-height: ${AvatarSize.tiny}px;
  `}
  ${props => props.size === 'small' && css`
    font-size: ${typography.size.s1}px;
    line-height: ${AvatarSize.small}px;
  `}
  ${props => props.size === 'large' && css`
    font-size: ${typography.size.s3}px;
    line-height: ${AvatarSize.large}px;
  `}
`;

interface AllyProps {
  [key : string]: boolean|string
}

/**
 * 头徽组件
 * @param props 
 */
export function Avatar(props: AvatarProps) {
  const { isLoading, username, src, size } = props;
  const avatarFigure = useMemo(()=>{
    let avatarDom = <Icon icon='useralt' />;
    const allyProps: AllyProps = {};
    if(isLoading) {
      allyProps['aria-busy'] = true;
      allyProps['aria-label'] = 'Loading avatar ...';
    } else if(src) {
      avatarDom = (<img src={src} alt={username} data-testid="avatar-img" />)
    } else {
      allyProps['aria-label'] = username!;
      avatarDom = (<Initial size={size} aira-hidden="true" data-testid="avatar-username">{username!.substring(0,1)}</Initial>);
    }
    return avatarDom;
  }, [isLoading, src, size, username]);

  return <Image size={size} isLoading={isLoading} src={src} data-testid="avatar-div" {...props}>{avatarFigure}</Image>
}

/**
 * 默认值数据
 */
Avatar.defaultProps = {
  size: 'medium',
  src: null,
  isLoading: false,
  username: 'loading'
};