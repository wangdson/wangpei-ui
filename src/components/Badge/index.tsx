import React, { PropsWithChildren, HTMLAttributes } from 'react';
import styled from 'styled-components';
import { color, background, typography } from "../shared/style";

/**
 * 徽章颜色
 */
export const badgeColor = {
  positive: color.positive,
  negative: color.negative,
  neutral: color.dark,
  warning: color.warning,
  error: color.lightest
}

/**
 * 徽章背景色
 */
export const badgeBackground = {
  positive: background.positive,
  negative: background.negative,
  neutral: color.mediumlight,
  warning: background.warning,
  error: color.negative
}

const BadgeWrapper = styled.div<BadgeProps>`
  display: inline-block;
  vertical-align: top;
  font-size: 11px;
  line-height: 12px;
  padding: 4px 12px;
  border-radius: 3em;
  font-weight: ${typography.weight.bold};
  color: ${props => badgeColor[props.status!]};
  background: ${props => badgeBackground[props.status!]};

  svg {
    width: 12px;
    height: 12px;
    margin-top: -2px;
    margin-right: 4px;
  }
`;

type Status = 'positive'|'negative'|'neutral'|'warning'|'error'; 
type StatusObj = {
  [key in Status]: Status
}

export const statusList:StatusObj = {
	'positive': 'positive',
	'negative':'negative',
	'neutral': 'neutral',
	'warning': 'warning',
	'error':'error'
}

/**
 * 徽章组件类型描述
 */
export interface BadgeProps extends HTMLAttributes<HTMLDivElement>{
  status?: Status; 
  className?: string;
}

/**
 * 徽章组件
 * @param props 
 */
export function Badge(props: PropsWithChildren<BadgeProps>){
  return <BadgeWrapper {...props} status={props.status}>{props.children}</BadgeWrapper>
}

Badge.defaultProps = {
  status: 'neutral'
}

export default Badge;