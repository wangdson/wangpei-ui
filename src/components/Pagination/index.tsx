import React, { PropsWithChildren, useEffect, useMemo, useState, CSSProperties } from 'react';
import { color, typography } from "../shared/style";
import { darken, rgba, opacify } from "polished";
import { easing } from "../shared/animation";
import Button from '../Button';
import { Icon } from '../Icon';
import { PageUl } from './style';
import { calculateMove } from './action';

export * from './Table';

export interface PaginationProps {
  /** 每页显示多少条数据 */
  pageSize?: number;
  /** 默认显示第几页 */
  defaultCurrent?: number;
  /** 总数据量 */
  total: number;
  /** 外层style*/
	style?:CSSProperties;
	/**外层类名 */
  classnames?:string;
  /** 分页最大显示长度 */
  maxPage?: number;
  /** 回调函数 */
  callback?: (v: number)=>void;
}


/**
 * 表格分页组件
 * @param props 
 */
export function Pagination(props: PropsWithChildren<PaginationProps>){
  const { pageSize, total, maxPage, callback, defaultCurrent, style, classnames } = props;
  const [current, setCurrent] = useState(defaultCurrent!);
  const [page, setPage] = useState<Array<number>>();
  const size = Math.ceil(total / pageSize!);
  const totalPage = useMemo(()=>{
    const size = Math.ceil(total / pageSize!);
    let tempPage;
    if(size > maxPage!) {
      tempPage = Array.from({length: maxPage!}).fill(1).map((x,y)=> y + 1);
    } else {
      tempPage = Array.from({length: size}).fill(1).map((x,y) => y + 1);
    }
    setPage(tempPage);
    // let arr = calculateMove(current, tempPage!, size);
    // if(arr) {
    //   setPage(arr);
    // }
    return size;
  },[total, pageSize, maxPage]);
  /** 当current有变化则返回回去 */
  useEffect(()=>{
    if(callback){
      callback(current);
    }
  }, [callback, current]);

  return (
    <PageUl style={style} className={classnames}>
      <li>
        <Button appearance="primaryOutline" disabled={current === 1 ? true: false} onClick={()=>{
          if(page && page.length) {
            if(page[0] > 1) {
              // 当前离第一页有偏移
              setCurrent((state)=>state-1);
              const mid = Math.floor(maxPage! / 2);
              if(current <= page[mid]) {
                let tempPage = page.map(x => x + current - page[mid] - 1)
                setPage(tempPage);
              }
            } else if(current !== page[0]) {
              // 当前位置不在第一位，但是当前页要往前进一位
              setCurrent((state)=>state-1);
            }
          }
        }}>
          <Icon icon="arrowleft"></Icon>
        </Button>
      </li>
      {page && page.map((item, i) => (
        <li key={i}>
          <Button appearance={item===current ? 'primary': 'primaryOutline'} onClick={()=>{
            setCurrent(item);
            let arr = calculateMove(item, page!, size);
            if(arr) {
              setPage(arr);
            }
          }}>
            {item}
          </Button>
        </li>
      ))}
      <li>
        <Button appearance="primaryOutline" disabled={current === totalPage ? true: false} onClick={()=>{
          if(page && page.length) {
            if(page[page.length-1] < totalPage){
              // 如果当前显示的数组最后一位不是最大页码
              setCurrent((state)=>state+1);
              const mid = Math.floor(maxPage! / 2);
              if(current >= page[mid]) {
                let tempPage = page.map(x=>x + current - page[mid] +  +1);
                setPage(tempPage);
              }
            } else if(current !== page[page.length-1]){
              // 判断当前页码是不是可显示页码列表的最后一页
              setCurrent((state)=>state+1);
            }
          }
        }}>
          <Icon icon="arrowright"></Icon>
        </Button>
      </li>
    </PageUl>
  );
}

Pagination.defaultProps = {
  total: 1000,
  pageSize: 10,
  maxPage: 5,
  defaultCurrent: 11
}

export default Pagination;