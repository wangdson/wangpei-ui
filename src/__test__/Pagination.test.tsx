import React from "react";
import { render } from "@testing-library/react";
import { Pagination, SourceDataType, Table } from "../components/Pagination";

const columns = [
	{
		title: "Name",
		dataIndex: "name",
	},
	{
		title: "Chinese Score",
		dataIndex: "chinese",
		sorter: {
			compare: (a: SourceDataType, b: SourceDataType) =>
				b.chinese - a.chinese,
		},
	},
	{
		title: "Math Score",
		dataIndex: "math",
		sorter: {
			compare: (a: SourceDataType, b: SourceDataType) => b.math - a.math,
		},
	},
	{
		title: "English Score",
		dataIndex: "english",
		sorter: {
			compare: (a: SourceDataType, b: SourceDataType) =>
				b.english - a.english,
		},
	},
];
const data = [
	{
		key: "1",
		name: "John Brown",
		chinese: 55,
		math: 60,
		english: 70,
	},
	{
		key: "2",
		name: "Jim Green",
		chinese: 98,
		math: 66,
		english: 89,
	},
	{
		key: "3",
		name: "Joe Black",
		chinese: 78,
		math: 90,
		english: 70,
	},
	{
		key: "4",
		name: "Jim Red",
		chinese: 88,
		math: 99,
		english: 89,
	},
];

describe('test Pagination component', ()=>{
  it('it render corrent Pagination', ()=>{
    const wrapper = render(<Pagination 
      pageSize={10}
      maxPage={5}
      total={1000}
      defaultCurrent={1}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('it render corrent Table', ()=>{
    const wrapper = render(<Table
      data={data}
      columns={columns}
      sorted={true}
      pageSize={10}
      pagination={false}
    />);
    expect(wrapper).toMatchSnapshot();
  });
})