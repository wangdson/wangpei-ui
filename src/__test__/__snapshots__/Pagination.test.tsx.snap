// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Pagination component it render corrent Pagination 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="sc-jSFkmK kQqwCS"
      >
        <li>
          <button
            class="sc-dlnjPT lnHCYJ"
            data-testid="button"
            disabled=""
          >
            <span
              class="sc-bdnylx bRlASU"
            >
              <svg
                class="sc-hKFyIo epBBzH"
                data-testid="icon-svg"
                height="20px"
                viewBox="0 0 1024 1024"
                width="20px"
              >
                <path
                  class="sc-eCApGN ijoIcd"
                  color="black"
                  d="M257.93 511.976c0-10.236 3.902-20.47 11.71-28.282l344.098-344.158c15.622-15.624 40.946-15.624 56.57-0.006 15.622 15.622 15.624 40.948 0.004 56.568l-315.82 315.876 315.868 315.922c15.618 15.624 15.618 40.952-0.004 56.568-15.622 15.62-40.95 15.618-56.57-0.006l-344.146-344.202c-7.808-7.81-11.71-18.044-11.71-28.28z"
                  data-testid="icon-path"
                />
              </svg>
            </span>
          </button>
        </li>
        <li>
          <button
            class="sc-dlnjPT fQqOyL"
            data-testid="button"
          >
            <span
              class="sc-bdnylx bRlASU"
            >
              1
            </span>
          </button>
        </li>
        <li>
          <button
            class="sc-dlnjPT hhkCkh"
            data-testid="button"
          >
            <span
              class="sc-bdnylx bRlASU"
            >
              2
            </span>
          </button>
        </li>
        <li>
          <button
            class="sc-dlnjPT hhkCkh"
            data-testid="button"
          >
            <span
              class="sc-bdnylx bRlASU"
            >
              3
            </span>
          </button>
        </li>
        <li>
          <button
            class="sc-dlnjPT hhkCkh"
            data-testid="button"
          >
            <span
              class="sc-bdnylx bRlASU"
            >
              4
            </span>
          </button>
        </li>
        <li>
          <button
            class="sc-dlnjPT hhkCkh"
            data-testid="button"
          >
            <span
              class="sc-bdnylx bRlASU"
            >
              5
            </span>
          </button>
        </li>
        <li>
          <button
            class="sc-dlnjPT hhkCkh"
            data-testid="button"
          >
            <span
              class="sc-bdnylx bRlASU"
            >
              <svg
                class="sc-hKFyIo epBBzH"
                data-testid="icon-svg"
                height="20px"
                viewBox="0 0 1024 1024"
                width="20px"
              >
                <path
                  class="sc-eCApGN ijoIcd"
                  color="black"
                  d="M768.072 514.022c0 10.236-3.904 20.47-11.712 28.282l-344.098 344.156c-15.62 15.624-40.946 15.624-56.568 0.006-15.622-15.622-15.624-40.948-0.006-56.568l315.82-315.876-315.868-315.922c-15.618-15.624-15.618-40.952 0.004-56.568 15.624-15.62 40.95-15.618 56.57 0.006l344.144 344.204c7.81 7.81 11.714 18.044 11.714 28.28z"
                  data-testid="icon-path"
                />
              </svg>
            </span>
          </button>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="sc-jSFkmK kQqwCS"
    >
      <li>
        <button
          class="sc-dlnjPT lnHCYJ"
          data-testid="button"
          disabled=""
        >
          <span
            class="sc-bdnylx bRlASU"
          >
            <svg
              class="sc-hKFyIo epBBzH"
              data-testid="icon-svg"
              height="20px"
              viewBox="0 0 1024 1024"
              width="20px"
            >
              <path
                class="sc-eCApGN ijoIcd"
                color="black"
                d="M257.93 511.976c0-10.236 3.902-20.47 11.71-28.282l344.098-344.158c15.622-15.624 40.946-15.624 56.57-0.006 15.622 15.622 15.624 40.948 0.004 56.568l-315.82 315.876 315.868 315.922c15.618 15.624 15.618 40.952-0.004 56.568-15.622 15.62-40.95 15.618-56.57-0.006l-344.146-344.202c-7.808-7.81-11.71-18.044-11.71-28.28z"
                data-testid="icon-path"
              />
            </svg>
          </span>
        </button>
      </li>
      <li>
        <button
          class="sc-dlnjPT fQqOyL"
          data-testid="button"
        >
          <span
            class="sc-bdnylx bRlASU"
          >
            1
          </span>
        </button>
      </li>
      <li>
        <button
          class="sc-dlnjPT hhkCkh"
          data-testid="button"
        >
          <span
            class="sc-bdnylx bRlASU"
          >
            2
          </span>
        </button>
      </li>
      <li>
        <button
          class="sc-dlnjPT hhkCkh"
          data-testid="button"
        >
          <span
            class="sc-bdnylx bRlASU"
          >
            3
          </span>
        </button>
      </li>
      <li>
        <button
          class="sc-dlnjPT hhkCkh"
          data-testid="button"
        >
          <span
            class="sc-bdnylx bRlASU"
          >
            4
          </span>
        </button>
      </li>
      <li>
        <button
          class="sc-dlnjPT hhkCkh"
          data-testid="button"
        >
          <span
            class="sc-bdnylx bRlASU"
          >
            5
          </span>
        </button>
      </li>
      <li>
        <button
          class="sc-dlnjPT hhkCkh"
          data-testid="button"
        >
          <span
            class="sc-bdnylx bRlASU"
          >
            <svg
              class="sc-hKFyIo epBBzH"
              data-testid="icon-svg"
              height="20px"
              viewBox="0 0 1024 1024"
              width="20px"
            >
              <path
                class="sc-eCApGN ijoIcd"
                color="black"
                d="M768.072 514.022c0 10.236-3.904 20.47-11.712 28.282l-344.098 344.156c-15.62 15.624-40.946 15.624-56.568 0.006-15.622-15.622-15.624-40.948-0.006-56.568l315.82-315.876-315.868-315.922c-15.618-15.624-15.618-40.952 0.004-56.568 15.624-15.62 40.95-15.618 56.57 0.006l344.144 344.204c7.81 7.81 11.714 18.044 11.714 28.28z"
                data-testid="icon-path"
              />
            </svg>
          </span>
        </button>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Pagination component it render corrent Table 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <table
          class="sc-gKAblj bpQvcg"
        >
          <thead>
            <tr>
              <td
                style="position: relative;"
              >
                <span>
                  Name
                </span>
              </td>
              <td
                style="position: relative;"
              >
                <span>
                  Chinese Score
                </span>
                <span
                  class="sc-iCoHVE bojTjA"
                >
                  <svg
                    class="sc-hKFyIo iDVKCX"
                    color="#666666"
                    data-testid="icon-svg"
                    height="20px"
                    viewBox="0 0 1024 1024"
                    width="20px"
                  >
                    <path
                      class="sc-eCApGN hqgtJJ"
                      color="#666666"
                      d="M512.024 256c10.236 0 20.47 3.904 28.282 11.712l344.154 344.098c15.624 15.62 15.624 40.946 0.006 56.57-15.622 15.622-40.948 15.624-56.568 0.004l-315.876-315.82-315.922 315.868c-15.624 15.618-40.952 15.618-56.568-0.004-15.62-15.624-15.618-40.95 0.006-56.57l344.204-344.144c7.81-7.81 18.046-11.714 28.282-11.714z"
                      data-testid="icon-path"
                    />
                  </svg>
                  <svg
                    class="sc-hKFyIo iDVKCX"
                    color="#666666"
                    data-testid="icon-svg"
                    height="20px"
                    viewBox="0 0 1024 1024"
                    width="20px"
                  >
                    <path
                      class="sc-eCApGN hqgtJJ"
                      color="#666666"
                      d="M511.976 768.002c-10.236 0-20.47-3.904-28.282-11.712l-344.154-344.098c-15.624-15.62-15.624-40.946-0.006-56.57 15.622-15.622 40.948-15.624 56.568-0.004l315.876 315.82 315.922-315.868c15.624-15.618 40.952-15.616 56.568 0.004 15.62 15.624 15.618 40.95-0.006 56.57l-344.204 344.144c-7.81 7.81-18.046 11.714-28.282 11.714z"
                      data-testid="icon-path"
                    />
                  </svg>
                </span>
              </td>
              <td
                style="position: relative;"
              >
                <span>
                  Math Score
                </span>
                <span
                  class="sc-iCoHVE bojTjA"
                >
                  <svg
                    class="sc-hKFyIo iDVKCX"
                    color="#666666"
                    data-testid="icon-svg"
                    height="20px"
                    viewBox="0 0 1024 1024"
                    width="20px"
                  >
                    <path
                      class="sc-eCApGN hqgtJJ"
                      color="#666666"
                      d="M512.024 256c10.236 0 20.47 3.904 28.282 11.712l344.154 344.098c15.624 15.62 15.624 40.946 0.006 56.57-15.622 15.622-40.948 15.624-56.568 0.004l-315.876-315.82-315.922 315.868c-15.624 15.618-40.952 15.618-56.568-0.004-15.62-15.624-15.618-40.95 0.006-56.57l344.204-344.144c7.81-7.81 18.046-11.714 28.282-11.714z"
                      data-testid="icon-path"
                    />
                  </svg>
                  <svg
                    class="sc-hKFyIo iDVKCX"
                    color="#666666"
                    data-testid="icon-svg"
                    height="20px"
                    viewBox="0 0 1024 1024"
                    width="20px"
                  >
                    <path
                      class="sc-eCApGN hqgtJJ"
                      color="#666666"
                      d="M511.976 768.002c-10.236 0-20.47-3.904-28.282-11.712l-344.154-344.098c-15.624-15.62-15.624-40.946-0.006-56.57 15.622-15.622 40.948-15.624 56.568-0.004l315.876 315.82 315.922-315.868c15.624-15.618 40.952-15.616 56.568 0.004 15.62 15.624 15.618 40.95-0.006 56.57l-344.204 344.144c-7.81 7.81-18.046 11.714-28.282 11.714z"
                      data-testid="icon-path"
                    />
                  </svg>
                </span>
              </td>
              <td
                style="position: relative;"
              >
                <span>
                  English Score
                </span>
                <span
                  class="sc-iCoHVE bojTjA"
                >
                  <svg
                    class="sc-hKFyIo iDVKCX"
                    color="#666666"
                    data-testid="icon-svg"
                    height="20px"
                    viewBox="0 0 1024 1024"
                    width="20px"
                  >
                    <path
                      class="sc-eCApGN hqgtJJ"
                      color="#666666"
                      d="M512.024 256c10.236 0 20.47 3.904 28.282 11.712l344.154 344.098c15.624 15.62 15.624 40.946 0.006 56.57-15.622 15.622-40.948 15.624-56.568 0.004l-315.876-315.82-315.922 315.868c-15.624 15.618-40.952 15.618-56.568-0.004-15.62-15.624-15.618-40.95 0.006-56.57l344.204-344.144c7.81-7.81 18.046-11.714 28.282-11.714z"
                      data-testid="icon-path"
                    />
                  </svg>
                  <svg
                    class="sc-hKFyIo iDVKCX"
                    color="#666666"
                    data-testid="icon-svg"
                    height="20px"
                    viewBox="0 0 1024 1024"
                    width="20px"
                  >
                    <path
                      class="sc-eCApGN hqgtJJ"
                      color="#666666"
                      d="M511.976 768.002c-10.236 0-20.47-3.904-28.282-11.712l-344.154-344.098c-15.624-15.62-15.624-40.946-0.006-56.57 15.622-15.622 40.948-15.624 56.568-0.004l315.876 315.82 315.922-315.868c15.624-15.618 40.952-15.616 56.568 0.004 15.62 15.624 15.618 40.95-0.006 56.57l-344.204 344.144c-7.81 7.81-18.046 11.714-28.282 11.714z"
                      data-testid="icon-path"
                    />
                  </svg>
                </span>
              </td>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                John Brown
              </td>
              <td>
                55
              </td>
              <td>
                60
              </td>
              <td>
                70
              </td>
            </tr>
            <tr>
              <td>
                Jim Green
              </td>
              <td>
                98
              </td>
              <td>
                66
              </td>
              <td>
                89
              </td>
            </tr>
            <tr>
              <td>
                Joe Black
              </td>
              <td>
                78
              </td>
              <td>
                90
              </td>
              <td>
                70
              </td>
            </tr>
            <tr>
              <td>
                Jim Red
              </td>
              <td>
                88
              </td>
              <td>
                99
              </td>
              <td>
                89
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <table
        class="sc-gKAblj bpQvcg"
      >
        <thead>
          <tr>
            <td
              style="position: relative;"
            >
              <span>
                Name
              </span>
            </td>
            <td
              style="position: relative;"
            >
              <span>
                Chinese Score
              </span>
              <span
                class="sc-iCoHVE bojTjA"
              >
                <svg
                  class="sc-hKFyIo iDVKCX"
                  color="#666666"
                  data-testid="icon-svg"
                  height="20px"
                  viewBox="0 0 1024 1024"
                  width="20px"
                >
                  <path
                    class="sc-eCApGN hqgtJJ"
                    color="#666666"
                    d="M512.024 256c10.236 0 20.47 3.904 28.282 11.712l344.154 344.098c15.624 15.62 15.624 40.946 0.006 56.57-15.622 15.622-40.948 15.624-56.568 0.004l-315.876-315.82-315.922 315.868c-15.624 15.618-40.952 15.618-56.568-0.004-15.62-15.624-15.618-40.95 0.006-56.57l344.204-344.144c7.81-7.81 18.046-11.714 28.282-11.714z"
                    data-testid="icon-path"
                  />
                </svg>
                <svg
                  class="sc-hKFyIo iDVKCX"
                  color="#666666"
                  data-testid="icon-svg"
                  height="20px"
                  viewBox="0 0 1024 1024"
                  width="20px"
                >
                  <path
                    class="sc-eCApGN hqgtJJ"
                    color="#666666"
                    d="M511.976 768.002c-10.236 0-20.47-3.904-28.282-11.712l-344.154-344.098c-15.624-15.62-15.624-40.946-0.006-56.57 15.622-15.622 40.948-15.624 56.568-0.004l315.876 315.82 315.922-315.868c15.624-15.618 40.952-15.616 56.568 0.004 15.62 15.624 15.618 40.95-0.006 56.57l-344.204 344.144c-7.81 7.81-18.046 11.714-28.282 11.714z"
                    data-testid="icon-path"
                  />
                </svg>
              </span>
            </td>
            <td
              style="position: relative;"
            >
              <span>
                Math Score
              </span>
              <span
                class="sc-iCoHVE bojTjA"
              >
                <svg
                  class="sc-hKFyIo iDVKCX"
                  color="#666666"
                  data-testid="icon-svg"
                  height="20px"
                  viewBox="0 0 1024 1024"
                  width="20px"
                >
                  <path
                    class="sc-eCApGN hqgtJJ"
                    color="#666666"
                    d="M512.024 256c10.236 0 20.47 3.904 28.282 11.712l344.154 344.098c15.624 15.62 15.624 40.946 0.006 56.57-15.622 15.622-40.948 15.624-56.568 0.004l-315.876-315.82-315.922 315.868c-15.624 15.618-40.952 15.618-56.568-0.004-15.62-15.624-15.618-40.95 0.006-56.57l344.204-344.144c7.81-7.81 18.046-11.714 28.282-11.714z"
                    data-testid="icon-path"
                  />
                </svg>
                <svg
                  class="sc-hKFyIo iDVKCX"
                  color="#666666"
                  data-testid="icon-svg"
                  height="20px"
                  viewBox="0 0 1024 1024"
                  width="20px"
                >
                  <path
                    class="sc-eCApGN hqgtJJ"
                    color="#666666"
                    d="M511.976 768.002c-10.236 0-20.47-3.904-28.282-11.712l-344.154-344.098c-15.624-15.62-15.624-40.946-0.006-56.57 15.622-15.622 40.948-15.624 56.568-0.004l315.876 315.82 315.922-315.868c15.624-15.618 40.952-15.616 56.568 0.004 15.62 15.624 15.618 40.95-0.006 56.57l-344.204 344.144c-7.81 7.81-18.046 11.714-28.282 11.714z"
                    data-testid="icon-path"
                  />
                </svg>
              </span>
            </td>
            <td
              style="position: relative;"
            >
              <span>
                English Score
              </span>
              <span
                class="sc-iCoHVE bojTjA"
              >
                <svg
                  class="sc-hKFyIo iDVKCX"
                  color="#666666"
                  data-testid="icon-svg"
                  height="20px"
                  viewBox="0 0 1024 1024"
                  width="20px"
                >
                  <path
                    class="sc-eCApGN hqgtJJ"
                    color="#666666"
                    d="M512.024 256c10.236 0 20.47 3.904 28.282 11.712l344.154 344.098c15.624 15.62 15.624 40.946 0.006 56.57-15.622 15.622-40.948 15.624-56.568 0.004l-315.876-315.82-315.922 315.868c-15.624 15.618-40.952 15.618-56.568-0.004-15.62-15.624-15.618-40.95 0.006-56.57l344.204-344.144c7.81-7.81 18.046-11.714 28.282-11.714z"
                    data-testid="icon-path"
                  />
                </svg>
                <svg
                  class="sc-hKFyIo iDVKCX"
                  color="#666666"
                  data-testid="icon-svg"
                  height="20px"
                  viewBox="0 0 1024 1024"
                  width="20px"
                >
                  <path
                    class="sc-eCApGN hqgtJJ"
                    color="#666666"
                    d="M511.976 768.002c-10.236 0-20.47-3.904-28.282-11.712l-344.154-344.098c-15.624-15.62-15.624-40.946-0.006-56.57 15.622-15.622 40.948-15.624 56.568-0.004l315.876 315.82 315.922-315.868c15.624-15.618 40.952-15.616 56.568 0.004 15.62 15.624 15.618 40.95-0.006 56.57l-344.204 344.144c-7.81 7.81-18.046 11.714-28.282 11.714z"
                    data-testid="icon-path"
                  />
                </svg>
              </span>
            </td>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              John Brown
            </td>
            <td>
              55
            </td>
            <td>
              60
            </td>
            <td>
              70
            </td>
          </tr>
          <tr>
            <td>
              Jim Green
            </td>
            <td>
              98
            </td>
            <td>
              66
            </td>
            <td>
              89
            </td>
          </tr>
          <tr>
            <td>
              Joe Black
            </td>
            <td>
              78
            </td>
            <td>
              90
            </td>
            <td>
              70
            </td>
          </tr>
          <tr>
            <td>
              Jim Red
            </td>
            <td>
              88
            </td>
            <td>
              99
            </td>
            <td>
              89
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
